_posts/2021-01-09-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,629 @@
+Indeed, Julia benchmarking tools don't count allocations on the stack as allocations at all.
+One can find extensive write ups of heap vs. stack allocations and how it works in general (though some mix the C specific factors with the CPU details).
+In Julia all mutable objects live are allocated on the heap.
+Until recently, immutable objects that contained references to heap allocated objects also had to live on the heap, i.e. only immutable objects with immutable fields (with immutable fields with...) could live on the stack.
+But with this change, all immutable objects can live on the stack, even if some of their fields live on the heap.
+An important consequence of this is that wrapper types, such as the `SubArray` returned from `@view x[1:2]`, now have no overhead to create.
+I find that in practice this often adds up to a 10-30% speed-up in real world code.
+
+(* It's actually really fast, but it is the kind of thing that rapidly adds up; and it is slow vs. operations that can happen without touching RAM.)
+
+
+## Pkg stdlib and the General Registry
+Writing this section is a bit hard as there is no NEWS.md nor HISTORY.md for the Pkg stdlib; and the general registry is more policy than software.
+However, some of the biggest changes have been in maturing out Pkg3, and its surroundings.
+
+
+### BinaryBuilder, Artifacts, Yggdasil and JLL packages
+
+This story was beginning to be told around Julia 1.0 time,
+but it wasn't really complete nor built into Pkg until Julia 1.3.
+You can read the documentation on [Artifact](https://julialang.github.io/Pkg.jl/v1/artifacts/), [BinaryBuilder](https://github.com/JuliaPackaging/BinaryBuilder.jl), and [Yggdasil](https://github.com/JuliaPackaging/Yggdrasil) for full details.
+This is the story about how Julia works with binary dependencies.
+As of now, it works really well.
+
+Sometimes people misunderstand the claims about Julia solving the two language problem as saying everything should be rewritten in Julia.
+Which is far from the truth, ain't noone got time for that.
+Julia has always had great foreign function interfacing (FFI); like [`ccall`](https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/) (as well as [PyCall](https://github.com/JuliaPy/PyCall.jl/), [RCall](https://github.com/JuliaInterop/RCall.jl), [JavaCall](https://github.com/JuliaInterop/JavaCall.jl), and a bunch of others, for non-C-style binaries).
+While FFI makes it easy to call binaries, how about actually getting them onto your machine?
+
+Before Julia 1.0, the standard was to run arbitrary code in the `deps/build.jl` file, which often used [BinDeps.jl](https://github.com/JuliaPackaging/BinDeps.jl/) to download and compile things.
+Around 1.0, this changed to having the `deps/build.jl` call [BinaryProvider.jl](https://github.com/JuliaPackaging/BinaryProvider.jl/) to download a compiled binary built with [BinaryBuilder.jl](https://github.com/JuliaPackaging/BinaryBuilder.jl), and store that alongside the package code.
```

Comment:
For the record, we have a website for BinaryBuilder: https://binarybuilder.org/!  It also shows an asciinema demo of the wizard.  You can maybe link to it instead of the GitHub repository?